tensordict/prototype/tensorclass.py

```diff
@@ -356,24 +413,52 @@ def _batch_size_setter(self, new_size: torch.Size) -> None:
 def __eq__(self, other):
     if not isinstance(other, self.__class__):
         return False
-    return self.tensordict == other.tensordict
+    keys1 = set(self.non_tensordict.keys())
+    keys2 = set(other.non_tensordict.keys())
+    if len(keys1.difference(keys2)) or len(keys1) != len(keys2):
+        raise KeyError(f"keys in {self} and {other} mismatch, got {keys1} and {keys2}")
+    non_tensor = {}
+    for key, value in self.non_tensordict.items():
+        non_tensor[key] = value == other.non_tensordict.get(key)
```

Comment:
why using the `.get(key)` method? the `[key]` call should be safe as key presence has already been checked